src/osd/PrimaryLogPG.h

```diff
@@ -840,6 +840,12 @@ class PrimaryLogPG : public PG, public PGBackend::Listener {
       // requeue at front of scrub blocking queue if we are blocked by scrub
       for (auto &&p: to_req) {
 	if (scrubber.write_blocked_by_scrub(p.first.get_head())) {
+          list<OpRequestRef>::iterator op_iter = p.second.begin();
+          while (op_iter != p.second.end()) {
+            (*op_iter)->mark_delayed("waiting for scrub");
+            ++op_iter;
+          }
```

Comment:
How about
<pre>
for (auto op : p.second) {
  op->mark_delayed("waiting for scrub");
}
</pre>
?
otherwise, lgtm!